addons/account_budget/account_budget.py

```diff
@@ -1,210 +1,162 @@
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as
-#    published by the Free Software Foundation, either version 3 of the
-#    License, or (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
-
-from datetime import date, datetime
-
-from openerp.osv import fields, osv
-from openerp.tools import ustr, DEFAULT_SERVER_DATE_FORMAT
-from openerp.tools.translate import _
+from openerp import api, fields, models, tools, _
+from openerp.tools import ustr
 from openerp.exceptions import UserError
 
-import openerp.addons.decimal_precision as dp
-
 
 # ---------------------------------------------------------
 # Utils
 # ---------------------------------------------------------
 def strToDate(dt):
-    return date(int(dt[0:4]), int(dt[5:7]), int(dt[8:10]))
+    return fields.Date.from_string(dt)
 
 def strToDatetime(strdate):
-    return datetime.strptime(strdate, DEFAULT_SERVER_DATE_FORMAT)
+    return fields.Datetime.from_string(strdate)
 
 # ---------------------------------------------------------
 # Budgets
 # ---------------------------------------------------------
-class account_budget_post(osv.osv):
+class AccountBudgetPost(models.Model):
     _name = "account.budget.post"
     _description = "Budgetary Position"
-    _columns = {
-        'code': fields.char('Code', size=64, required=True),
-        'name': fields.char('Name', required=True),
-        'account_ids': fields.many2many('account.account', 'account_budget_rel', 'budget_id', 'account_id', 'Accounts', domain=[('deprecated', '=', False)]),
-        'crossovered_budget_line': fields.one2many('crossovered.budget.lines', 'general_budget_id', 'Budget Lines'),
-        'company_id': fields.many2one('res.company', 'Company', required=True),
-    }
-    _defaults = {
-        'company_id': lambda self, cr, uid, c: self.pool.get('res.company')._company_default_get(cr, uid, 'account.budget.post', context=c)
-    }
+
+    code = fields.Char(required=True)
+    name = fields.Char(required=True)
+    account_ids = fields.Many2many('account.account', 'account_budget_rel', 'budget_id', 'account_id', string='Accounts', domain=[('deprecated', '=', False)])
+    crossovered_budget_line = fields.One2many('crossovered.budget.lines', 'general_budget_id', string='Budget Lines')
+    company_id = fields.Many2one('res.company', string='Company', required=True, default=lambda self: self.env.user.company_id)
```

Comment:
Use:
default=lambda self: self.env['res.company']._company_default_get('account.budget.post')
for default value.